presto-parquet/src/main/java/io/prestosql/parquet/writer/ParquetSchemaConverter.java

```diff
@@ -14,53 +14,42 @@
 package io.prestosql.parquet.writer;
 
 import com.google.common.collect.ImmutableList;
-import io.prestosql.spi.PrestoException;
 import io.prestosql.spi.type.ArrayType;
-import io.prestosql.spi.type.CharType;
-import io.prestosql.spi.type.DecimalType;
 import io.prestosql.spi.type.MapType;
-import io.prestosql.spi.type.RealType;
 import io.prestosql.spi.type.RowType;
 import io.prestosql.spi.type.Type;
-import io.prestosql.spi.type.VarbinaryType;
-import io.prestosql.spi.type.VarcharType;
 import org.apache.parquet.schema.GroupType;
 import org.apache.parquet.schema.MessageType;
-import org.apache.parquet.schema.OriginalType;
-import org.apache.parquet.schema.PrimitiveType;
 import org.apache.parquet.schema.Types;
 
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-import static io.prestosql.spi.StandardErrorCode.NOT_SUPPORTED;
-import static io.prestosql.spi.type.BigintType.BIGINT;
-import static io.prestosql.spi.type.BooleanType.BOOLEAN;
-import static io.prestosql.spi.type.DateType.DATE;
-import static io.prestosql.spi.type.DoubleType.DOUBLE;
-import static io.prestosql.spi.type.IntegerType.INTEGER;
-import static io.prestosql.spi.type.SmallintType.SMALLINT;
 import static io.prestosql.spi.type.StandardTypes.ARRAY;
 import static io.prestosql.spi.type.StandardTypes.MAP;
 import static io.prestosql.spi.type.StandardTypes.ROW;
-import static io.prestosql.spi.type.TimestampType.TIMESTAMP;
-import static io.prestosql.spi.type.TinyintType.TINYINT;
-import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 import static org.apache.parquet.Preconditions.checkArgument;
 import static org.apache.parquet.schema.Type.Repetition.OPTIONAL;
 
 public class ParquetSchemaConverter
 {
-    private Map<List<String>, Type> primitiveTypes = new HashMap<>();
+    private final PrimitiveTypeConverter primitiveTypeConverter;
+    private final Map<List<String>, Type> primitiveTypes = new HashMap<>();
     private final MessageType messageType;
 
     public ParquetSchemaConverter(List<Type> types, List<String> columnNames)
```

Comment:
It's better if `presto-parquet` doesn't have Hive specific logic. Remove this constructor variant, then you can move `HiveParquetPrimitiveTypeConverter` to the `presto-hive` module.

Reply:
Done.